### PR TITLE
Wolves

### PR DESCRIPTION
**Pull request by ghost**
PR 1 of how many of adding snowmap

Adds wolves, and pointed_at functionality for mobs

Wolves can be tamed, and commanded to do certain things depending on what you point at.

:cl:

* rscadd: Adds wolves

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+	icon_state = "wolf"
+	icon_living = "wolf"
+	icon_dead = "wolf_dead"
+	speak_chance = 5
+	turns_per_move = 4
+	response_help = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm = "hits"
+
+	speed = -1
```
Comment (9600bauds) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm:39]: maxresdefault

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+				if(istype(potential_pack.pack_alpha, /mob/living/simple_animal/hostile/wolf))
+					var/mob/living/simple_animal/hostile/wolf/alpha_challenger = potential_pack.pack_alpha
+					alpha_challenger.challenge(src)
+					challenge(alpha_challenger)
+
+/mob/living/simple_animal/hostile/wolf/CanAttack(var/atom/the_target)
+	//WE DON'T ATTACK INVULNERABLE MOBS (such as etheral jaunting mobs, or passengers of the adminbus)
+	if(the_target.flags & INVULNERABLE)
+		return 0
+	if(istype(the_target, /mob/dead/observer))
```
Comment (9600bauds) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm:100]: Should probably just cast the_target as /mob/living and do !istype() here

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+		return	//As is said in carp.dm, a bit of a hacky way of making something follow somebody
+	var/list/can_see = view(src, vision_range)
+	for(var/mob/living/simple_animal/hostile/wolf/potential_pack in can_see)
+		if(potential_pack.pack_alpha == src) //Source is alpha
+			if(!alpha_challenge)//If we're not in a duel
+				alpha_target = target
+
+		if(potential_pack.pack_alpha == pack_alpha) //Part of the same pack
+			potential_pack.target = target
+	if(ismob(target))
```
Comment (9600bauds) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm]: isliving

Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/wolf.dm]: Good catch

File: code/modules/mob/living/simple_animal/hostile/wolf.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/wolf.dm:39]: Would you prefer slow attack dogs? They're still slower than a well fed person

File: code/modules/mob/living/simple_animal/hostile/wolf.dm [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+			playsound(get_turf(src),'sound/items/eatfood.ogg', rand(10,50), 1)
+			visible_message("<span class='info'>\The [src] gobbles up \the [W]!")
+			nutrition += 15
+			if(prob(25))
+				if(!pack_alpha)
+					pack_alpha = user
+					to_chat(user, "<span class='info'>You have gained \the [src]'s trust.</span>")
+					var/n_name = copytext(sanitize(input(user, "What would you like to name your new friend?", "Wolf Name", null) as text|null), 1, MAX_NAME_LEN)
+					if(n_name && !user.incapacitated())
+						name = "[n_name]"
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/hostile/wolf.dm]: There's no need to wrap another string like this. Just do `name = n_name`.

Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/wolf.dm]: Sorry, force of habit

File: code/modules/mob/living/simple_animal/hostile/wolf.dm (hunk 1 above)
Comment (9600bauds) [reply code/modules/mob/living/simple_animal/hostile/wolf.dm:39]: -1 mob speed is still slower than a human? That's terrible

Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/wolf.dm:39]: You need to use move speed modifier

Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/wolf.dm:39]: Sentients are OP

**Comment by SkowronX** [comment]
nopics

**Comment by ghost** [comment]
Pics when im back

**Comment by ghost** [comment]
Just had an idea on how to rehandle how they feed at corpses by using butchering. Gonna do that when I get back

**Comment by mrpain666** [comment]
WOLVES HUNT IN PACKS

**Comment by Shadowmech88** [comment]
Are they weak to fire? They should be.

**Comment by ghost** [comment]
They're not weak to fire as that's the most common damage type in the game.

They can and may hunt in packs however, the framework's there, could even add a spawner for it.

**Comment by Shadowmech88** [comment]
What about making them hate fire?
Like reluctant to get close to campfires or flamethrower plumes.

**Comment by ghost** [comment]
Now that I can do

**Comment by ghost** [comment]
https://pomf.pyonpyon.moe/wgdvmg.webm

**Comment by ghost** [comment]
The pointing stuff is a bit off, in that it'll sometimes get caught up in itself and there's some natural delay

**Comment by Exxion** [comment]
Point hotkey when

**Comment by Shadowmech88** [comment]
We could make it be middle-click because who honestly switches hands with middle-click?

**Comment by Exxion** [comment]
Like four people, but they get REALLY UPSET when we talk about removing middle click hand switch

**Comment by PJB3005** [comment]
Rebindable keys when.

O right when BYOND's macro API stops being 💩 

Remake when?

**Comment by unid15** [comment]
@Exxion how the heck do you switch hands without middle click?

**Comment by jknpj** [comment]
@unid15 with X

**Comment by Shadowmech88** [comment]
Page up and Ctrl+S also work.

**Comment by Intigracy** [comment]
I use page up as well.

Only losers use middle click.

**Comment by Probe1** [comment]
I bet you dont use hotkey mode you twink

**Comment by Intigracy** [comment]
Only when I'm equipping stuff in a pile on the floor.
Do something about it, faggot.

**Comment by Probe1** [comment]
I will I'll robust your ass oh wait you never play nerd.

**Pull request merged**